sdks/python/apache_beam/runners/portability/fn_api_runner_test.py

```diff
@@ -47,12 +48,45 @@ def test_combine_per_key(self):
     pass
 
   def test_pardo_side_inputs(self):
-    # TODO(BEAM-1348): Enable once side inputs are supported in fn API.
-    pass
-
-  def test_pardo_unfusable_side_inputs(self):
-    # TODO(BEAM-1348): Enable once side inputs are supported in fn API.
-    pass
+    def cross_product(elem, sides):
+      for side in sides:
+        yield elem, side
+    with self.create_pipeline() as p:
+      main = p | 'main' >> beam.Create(['a', 'b', 'c'])
+      side = p | 'side' >> beam.Create(['x', 'y'])
+      assert_that(main | beam.FlatMap(cross_product, beam.pvalue.AsList(side)),
+                  equal_to([('a', 'x'), ('b', 'x'), ('c', 'x'),
+                            ('a', 'y'), ('b', 'y'), ('c', 'y')]))
+
+      # Now with some windowing.
+      pcoll = p | beam.Create(range(10)) | beam.Map(
+          lambda t: window.TimestampedValue(t, t))
+      # Intentionally choosing non-aligned windows to highlight the transition.
+      main = pcoll | 'WindowMain' >> beam.WindowInto(window.FixedWindows(5))
+      side = pcoll | 'WindowSide' >> beam.WindowInto(window.FixedWindows(7))
+      res = main | beam.Map(lambda x, s: (x, sorted(s)),
+                            beam.pvalue.AsList(side))
+      assert_that(
+          res,
+          equal_to([
+              # The window [0, 5) maps to the window [0, 7).
+              (0, range(7)),
+              (1, range(7)),
+              (2, range(7)),
+              (3, range(7)),
+              (4, range(7)),
+              # The window [5, 10) maps to the window [7, 14).
+              (5, range(7, 10)),
+              (6, range(7, 10)),
+              (7, range(7, 10)),
+              (8, range(7, 10)),
+              (9, range(7, 10))]),
+          label='windowed')
+
+
+#   def test_pardo_unfusable_side_inputs(self):
```

Comment:
<!--new_thread; commit:202dfc74e58e9736bb40ed0382fa03dd072b7ab4; resolved:0-->
Do you want to remove this test or implement it?

Reply:
I want to inherit the test that was formerly disabled...